File: src/com/huhx0015/datastructures/CustomLinkedList.java
package com.huhx0015.datastructures;

/**
 * Created by Michael Yoon Huh on 5/29/2017.
 */
public class CustomLinkedList {

    private Object data;
    private CustomLinkedList head;
    private CustomLinkedList tail;

    public CustomLinkedList(Object data) {
        this.data = data;
        this.head = null;
        this.tail = null;
    }

    public CustomLinkedList(Object data, CustomLinkedList head, CustomLinkedList tail) {
        this.data = data;
        this.head = head;
        this.tail = tail;
    }

    public Object getData() {
        return data;
    }

    public void setData(Object data) {
        this.data = data;
    }

    public CustomLinkedList getHead() {
        return head;
    }

    public void setHead(CustomLinkedList head) {
        this.head = head;
    }

    public CustomLinkedList getTail() {
        return tail;
    }

    public void setTail(CustomLinkedList tail) {
        this.tail = tail;
    }
}

File: src/main/com/huhx0015/algorithmtester/twentyseventeen/datastructures/CustomLinkedList.kt
package com.huhx0015.algorithmtester.twentyseventeen.datastructures

/**
 * Created by Michael Yoon Huh on 5/29/2017.
 */
class CustomLinkedList {

    var data: Any? = null
    var head: CustomLinkedList? = null
    var tail: CustomLinkedList? = null

    constructor(data: Any) {
        this.data = data
        this.head = null
        this.tail = null
    }

    constructor(data: Any, head: CustomLinkedList, tail: CustomLinkedList) {
        this.data = data
        this.head = head
        this.tail = tail
    }
}